include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -100,39 +141,67 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 
 	// main methods
 
-	//! Warning `outAddresses` needs to be primed with `invalid_value` values, otherwise no allocation happens for elements not equal to `invalid_value`
-	inline void multi_allocate(uint32_t binding, uint32_t count, value_type* outAddresses)
+#ifdef _NBL_DEBUG
+	std::unique_lock<std::recursive_mutex> stAccessVerifyDebugGuard()
 	{
-		#ifdef _NBL_DEBUG
 		std::unique_lock<std::recursive_mutex> tLock(stAccessVerfier,std::try_to_lock_t());
 		assert(tLock.owns_lock());
-		#endif // _NBL_DEBUG
+		return tLock;
+	}
+#else
+	bool stAccessVerifyDebugGuard() { return false; }
+#endif
+
+	//! Warning `outAddresses` needs to be primed with `invalid_value` values, otherwise no allocation happens for elements not equal to `invalid_value`
+	inline void multi_allocate(uint32_t binding, size_type count, value_type* outAddresses)
+	{
+		auto debugGuard = stAccessVerifyDebugGuard();
 
 		auto allocator = getBindingAllocator(binding);
-		for (uint32_t i=0; i<count; i++)
+		for (size_type i=0; i<count; i++)
 		{
 			if (outAddresses[i]!=AddressAllocator::invalid_address)
 				continue;
 
 			outAddresses[i] = allocator->alloc_addr(1,1);
+			// TODO: should also write something to the descriptor set (or probably leave that to the caller?)
```

Comment:
should go outside the loop and happen all at once in bulk.

reserve a vector of `SDescriptorInfo` for one descriptor write and fire it off

Reply:
ok in-case I wasn't clear, yes... leave that to the caller